sql/src/main/java/io/crate/analyze/relations/UnionSelect.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Licensed to Crate under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.  Crate licenses this file
+ * to you under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.analyze.relations;
+
+import io.crate.analyze.Fields;
+import io.crate.analyze.QuerySpec;
+import io.crate.analyze.symbol.Field;
+import io.crate.exceptions.ColumnUnknownException;
+import io.crate.metadata.Path;
+import io.crate.metadata.table.Operation;
+import io.crate.sql.tree.QualifiedName;
+
+import javax.annotation.Nonnull;
+import java.util.ArrayList;
+import java.util.List;
+
+public class UnionSelect implements QueriedRelation {
+
+    private final boolean distinct;
+    private final QuerySpec querySpec = new QuerySpec();
+    private final Fields fields;
+    private QueriedRelation left;
+    private QueriedRelation right;
+    private QualifiedName name;
+
+    public UnionSelect(QueriedRelation left, QueriedRelation right, boolean distinct) {
+        this.left = left;
+        this.right = right;
+        this.distinct = distinct;
+        this.name = left.getQualifiedName();
+
+        List<Field> fieldsFromLeft = left.fields();
+        fields = new Fields(fieldsFromLeft.size());
+        for (Field field : fieldsFromLeft) {
+            fields.add(field.path(), new Field(this, field.path(), field.valueType()));
+        }
+        querySpec.outputs(new ArrayList<>(fields.asList()));
+    }
+
+    public boolean isDistinct() {
```

Comment:
I'd say because we implemented parser support for the union and have this information available anyway. You're right the analyzer doesn't really need to know that yet. I'd be ok with removing it.

Reply:
I agree with @seut to remove it since as he pointed out we don't even use it in toString() currently. So when we implement the UNION [DISTINCT] we don't forget this for example.